source/dshow-settings.cpp

```diff
@@ -0,0 +1,398 @@
+#include <mutex>
+#include <fstream>
+#include <sstream>
+#include <iomanip>
+#include <comdef.h>
+#include "dshow-settings.hpp"
+
+namespace DShow {
+    void SetDeviceSettings(IEnumMoniker *enumMoniker, nlohmann::json &camSettings) {
+        ComPtr<IMoniker> moniker;
+        ComPtr<IAMVideoProcAmp> videoProcSettings;
+        ComPtr<IAMCameraControl> camControlSettings;
+        ComPtr<IPropertyBag> propertyBag;
+        HRESULT hr;
+
+        if (enumMoniker == nullptr) {
+            return;
+        }
+
+        while (enumMoniker->Next(1, &moniker, nullptr) == S_OK) {
+            hr = moniker->BindToStorage(nullptr, nullptr, IID_IPropertyBag, (void **)&propertyBag);
+            if (FAILED(hr)) {
+                continue;
+            }
+
+            VARIANT path, name;
+            path.vt = VT_BSTR;
+            name.vt = VT_BSTR;
+
+            hr = propertyBag->Read(L"FriendlyName", &name, nullptr);
+            if (SUCCEEDED(hr)) {
+                const std::string nameStr(_bstr_t(name.bstrVal, true));
+
+                hr = propertyBag->Read(L"DevicePath", &path, nullptr);
+                if (SUCCEEDED(hr)) {
+                    const std::string pathStr(_bstr_t(path.bstrVal, true));
+
+                    auto settings = std::find_if(camSettings.begin(), camSettings.end(), [nameStr, pathStr](const nlohmann::json& j) {
+                        return j.is_object() and j.value("DevicePath", "") == pathStr and j.value("DeviceName", "") == nameStr;
```

Comment:
```suggestion
                        return j.is_object() && j.value("DevicePath", "") == pathStr and j.value("DeviceName", "") == nameStr;
```
IMHO, better use canonical && for C++ code

Reply:
Changed to && here